src/transforms/remap.rs

```diff
@@ -650,6 +659,72 @@ fn push_dropped(
     output.push_named(DROPPED, event)
 }
 
+/// If the VRL returns a value that is not an array (see [`merge_array_definitions`]),
+/// or an object, that data is moved into the `message` field.
+fn move_fields_into_message(mut definition: schema::Definition) -> schema::Definition {
```

Comment:
I don't think this function name is totally accurate since we're just moving definitions, not the actual fields?